### PR TITLE
systemcmds/reboot: Fix reboot command for kernel mode

### PR DESCRIPTION
**Pull request by pussuw**
Fixes reboot command by:
- Start user workqueue thread in kernel mode by "lazy init"
- Provide hartid in a0 for __start() as required

### CODE REVIEW

File: platforms/nuttx/src/px4/microchip/mpfs/board_reset/board_reset.cpp [hunk 1]
```diff
@@ -50,9 +52,19 @@ static void board_reset_enter_bootloader()
...
 }
 
-int board_reset(int status)
+static void board_reset_enter_app(uintptr_t hartid)
 {
+	register long r0 asm("a0") = (long)(hartid);
+
+	asm volatile
+	(
+		"tail __start\n" :: "r"(r0) : "memory"
```
Comment (pussuw) [inline platforms/nuttx/src/px4/microchip/mpfs/board_reset/board_reset.cpp:61]: Need to disable interrupts here, will add patch for that soon

File: platforms/common/shutdown.cpp [hunk 2]
```diff
@@ -36,6 +36,11 @@
  * Implementation of the API declared in px4_shutdown.h.
  */
 
+#ifdef __PX4_NUTTX
+// Make sure this is defined, this is a kernel module
+#define __KERNEL__
```
Comment (jlaitine) [inline platforms/common/shutdown.cpp:41]: why not just add something like "target_compile_options(mmymodule.... PRIVATE -D__KERNEL__)" into CMakeLists.txt for the module, which compiles this and goes into kernel? This is fine as well, don't really care :)

Comment (pussuw) [reply platforms/common/shutdown.cpp:41]: Yes this is the preferred way but in this particular case, the library that takes shutdown.cpp is px4_platform and it is unambiguous where it goes. It is still not obvious / clear to me how that library should be handled. It is still just a massive mess in my head.

Comment (pussuw) [reply platforms/common/shutdown.cpp]: I separated shutdown into its own target, now its compile time flags can be adjusted

**Comment by pussuw** [comment]
What I was wondering is what the shutdown hooks in shutdown.cpp are for. Only reference I can find is this:

`px4_register_shutdown_hook(&Logger::request_stop_static);`

**Review by jlaitine — APPROVED** [review]

**Review by jlaitine — APPROVED** [review]
LGTM

**Pull request merged**